crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1168,6 +1168,15 @@ impl CheckpointBuilder {
         let mut batch = self.tables.checkpoint_content.batch();
         let mut all_tx_digests =
             Vec::with_capacity(new_checkpoints.iter().map(|(_, c)| c.size()).sum());
+
+        let last_certified = self
+            .tables
+            .certified_checkpoints
+            .keys()
+            .skip_to_last()
+            .next()
+            .transpose()?;
```

Comment:
I think it would be better to use the watermark (search for `CheckpointWatermark::HighestVerified`), since that proves that state sync is aware of the checkpoint, which in turn guarantees it can be propagated to other peers.. (I am pretty sure that state sync is the only way the certified_checkpoints table could get ahead of local certification anyway, but it still seems better to check directly).

Reply:
Also, you could give `SubmitCheckpointToConsensus` a handle to CheckpointStore, and do this check in there. I think that would probably result in less code.

Reply:
Nice that is much better! I will make that change but wanted to check if I should change it here as well (its where I stole this code from in the first place)

https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/checkpoints/mod.rs#L2303-L2318

Reply:
Change to using the watermark for ignoring incoming signatures seems appropriate so moved ahead with it